### PR TITLE
Move icu_num_utils to utils/num

### LINKED ISSUE #167: FixedDecimal as a standalone crate
This topic is a spin-off from my review of num-util in #141.

While reviewing the code I started reading more about other Rust crates that are working with numbers and their conversions. In particular, parsing and stringification.

My understanding is that `FixedDecimal` is a representation of a number which the user can perform certain operations on, such as elevating/lowering precision, adjusting visible digits, rounding etc.
Such number can be then stringified using the new information.

A canonical example of such use case is ECMA402 NumberFormat where the user may want to do:

```js
let nf = new Intl.NumberFormat(undefined, {
  minimumFractionDigits: 2,
});
nf.format(2.0); // "2.00"
```

And in Rust, FixedDecimal can be used for that purpose, like this:

```rust
// See https://github.com/unicode-org/icu4x/issues/167#issuecomment-654449715 for Shane's
// explanation how to use the API - the example here is *not* how he envisions it.

struct Number {
    integer: FixedDecimal,
    fraction: Option<FixedDecimal>
}

pub fn format<I: TryInto<Number>>(input: I, options: NumberFormatOptions) -> Result<String, _> {
    let mut num: Number = input.try_into()?;
    if let Some(fraction) = num.fraction {
        let shift = options.minfd - fraction.magnitude_range().end() - 1; // 2 - 0 - 1
        if shift > 0 {
            fraction.adjust_range(shift); // 1
        }
    }
    Ok(num.to_string())
}
```

This is of course oversimpified, uses allocation instead of Writer etc., but I tried to base this API on #141.

If this is, indeed, its main use case, then it seems to me that this structure is actually a fairly generic foundational number parsing/adjusting/serializing API that may be useful way beyond ICU4X.

In particular, as Shane pointed out in #166, some version of serialization of integers/floats is needed, and some version of parsing of integers/floats is needed.

The only thing beyond parsing/serializing is the concept of adjusting which digits are visible in both ways - by hiding (and rounding) digits, and by adding visible digits (such as trailing or leading `0`s).

We may want to allow for parsing from String where `2.00` could be parsed to our example `Number` with information about two fraction digits, while it cannot be parsed to any float type without losing that information.
We may also want to allow for serialization to Writer/String which should preserve the trailing zeros, or even allow the user to adjust the visible fraction digits and serialize as `2.000` or `2.0`.

Since those two operations require special structure to contain that information in an efficient way, and the usefulness of this utility goes beyond Internationalization domain, I'd like to put to consideration spawning this off of ICU4X as a standalone crate.

Such crate would provide `FixedDecimal` and maybe `FixedNumber` with parsing/serializing and rounding/visibility methods.

The benefits of such move would be:
* Separation of concerns
* Lower entry barrier for contributors
* Rally contributors with experience in number operations, math and Rust
* Much faster path to release and stability and in result feedback/usage
* Test-bed for our release process before ICU4X gets to that level

The potential downsides I see:
* Additional project management overhead from a separate repo/crate
* We'll need to ensure that such API will be usable for `PluralOperands` conversion

@sffc @Manishearth others - thoughts?

### PR DESCRIPTION
**Pull request by zbraniecki**
Fixes #167.

This just moves the `icu_num_utils` component to be `utils/num` and renames it to `icu_utils_num`.

I'm open to discuss whether the latter is necessary. My motivation is that it allows us to collect all utils as `icu_utils` and `num_utils` is just one of them.

### CODE REVIEW

**Comment by zbraniecki** [comment]
An open question is whether we want to have `utils/fixed_decimal` and `icu-utils-fixed-decimal` (or just `fixed-decimal`)? I'm not sure how many utils for numbers will we have and is `utils/num` and `icu-utils-num` just an alias for `fixed-decimal` or for a collection of APIs that `fixed-decimal` is just one of.

**Comment by sffc** [comment]
+1 for `utils/fixed_decimal`.

The package name should be either "icu-fixed-decimal" or "fixed-decimal", whichever makes you happier.

**Comment by zbraniecki** [comment]
great! I named it `fixed-decimal` in line with other crates, and if we decide to go with underscores in #304, we'll rename it together with all other crates.

File: utils/fixed-decimal/src/lib.rs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-//! `icu-num-util` is one of the [`ICU4X`] components.
+//! `fixed-decimal` is an utility crate of the [`ICU4X`] project.
```
Comment (sffc) [inline utils/fixed-decimal/src/lib.rs:1]: optional: s/an/a

https://www.google.com/search?q=%22an+utility%22+or+%22a+utility%22

**Review by sffc — DISMISSED** [review]
Thanks!

**Pull request merged**